resources/splunk_install.rb

```diff
@@ -0,0 +1,219 @@
+# Encoding: utf-8
+
+class SplunkInstall < ChefCompat::Resource
+  include PlatformHelpers, PathHelpers
+
+  resource_name :splunk_install
+
+  property :name, String, name_property: true, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :service_user, String, default: 'splunk'
+  property :service_password, String
+  property :version, String, required: true
+  property :build, String, required: true
+  property :source_url, String, default: 'https://download.splunk.com/products'
```

Comment:
It feels like it could be any of the three, but it might be leaning towards a base URL.